translate/storage/jsonl10n.py

```diff
@@ -246,3 +246,56 @@ def nested_set(target, path, value):
         out.write(json.dumps(units, separators=(',', ': '),
                              indent=4, ensure_ascii=False).encode(self.encoding))
         out.write(b'\n')
+
+
+class WebExtensionJsonUnit(base.TranslationUnit):
+    def __init__(self, source=None, ref=None, item=None):
+        if ref:
+            self._node = ref
+            self.notes = ref['description']
+            self._target = ref['message']
+        else:
+            self._node = OrderedDict((('message', ''), ('description', '')))
+        super(WebExtensionJsonUnit, self).__init__(source)
+
+    def setid(self, value):
+        self._id = value
+
+    def getid(self):
+        return self._id
+
+    def getlocations(self):
+        return [self.getid()]
+
+    def settarget(self, target):
+        super(WebExtensionJsonUnit, self).settarget(target)
+        self._node['message'] = target
+
+    def addnote(self, text, origin=None, position="append"):
+        super(WebExtensionJsonUnit, self).addnote(text, origin, position)
```

Comment:
I didn't want to copy much of the code from the base class, that's why I call super. Another option is to reimplement all logic behind target/notes handling and directly manipulate with inside dict instead of using properties defined by TranslationUnit...